src/quest.cpp

```diff
@@ -706,18 +713,26 @@ bool Quest_Context::DoAction(const EOPlus::Action& action)
 	else if (function_name == "settitle")
 	{
 		this->character->title = std::string(action.expr.args[0]);
+		if (this->character->title.length() > 32)
```

Comment:
I don't think we need to do these length checks upfront. If substr len is greater than the length of the string, you just get the whole string.

Reply:
Probably should have looked this up before assuming it behaved like C#'s where if you go out of bounds it throws an exception. Good catch